### PR TITLE
[TRA-466] add individual parameters of vaults

### PR DESCRIPTION
**Pull request by tqin7**
### Changelist
add individual parameters of vaults, which only has `lagged_price` for now.

### Test Plan
unit tests

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
  - Introduced `VaultParams` to manage parameters for each vault, including a new field `laggedPrice`.

- **Enhancements**
  - Enhanced initialization to set vault parameters during genesis initialization.
  - Added functions for retrieving and setting vault parameters with validation checks.

- **Bug Fixes**
  - Improved error handling in vault parameter functions.

- **Tests**
  - Added unit tests to ensure proper setting and retrieval of vault parameters.

### CODE REVIEW

File: protocol/x/vault/genesis.go [hunk 1]
```diff
@@ -24,7 +26,14 @@ func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState)
...
 			}
 		}
+		if vault.VaultParams != nil {
+			if err := k.SetVaultParams(ctx, *vault.VaultId, *vault.VaultParams); err != nil {
+				panic(err)
+			}
+		}
 	}
+
+	fmt.Println("init vault genesis", ExportGenesis(ctx, k))
```
Comment (vincentwschau) [inline protocol/x/vault/genesis.go:36]: nit: remove debug logging statement.

File: protocol/x/vault/keeper/params.go [hunk 2]
```diff
@@ -33,3 +34,40 @@ func (k Keeper) SetParams(
...
+	if b == nil {
+		return vaultParams, false
+	}
+
+	k.cdc.MustUnmarshal(b, &vaultParams)
+	return vaultParams, true
+}
+
+// SetVaultParams sets `VaultParams` in state for a given vault.
+// Returns an error iff validation fails.
```
Comment (vincentwschau) [inline protocol/x/vault/keeper/params.go:58]:
```suggestion
// Returns an error if validation fails.
```

File: protocol/x/vault/keeper/params_test.go [hunk 3]
```diff
@@ -46,3 +48,33 @@ func TestGetSetParams(t *testing.T) {
 	require.Error(t, err)
 	require.Equal(t, newParams, k.GetParams(ctx))
 }
+
+func TestGetSetVaultParams(t *testing.T) {
```
Comment (vincentwschau) [inline protocol/x/vault/keeper/params_test.go:52]: nit: Test setting params for non-existent vault.

**Review by vincentwschau — APPROVED** [review]

File: protocol/x/vault/keeper/params_test.go (hunk 3 above)
Comment (tqin7) [reply protocol/x/vault/keeper/params_test.go:52]: Could you clarify what this means? setting params for a vault that has no shares?

**Pull request merged**